synapse/storage/schema/__init__.py

```diff
@@ -35,7 +33,7 @@
 """
 
 
-SCHEMA_COMPAT_VERSION = 59
+SCHEMA_COMPAT_VERSION = 60  # 60: "outlier" not in internal_metadata.
```

Comment:
strictly speaking, the change to schema version 60 came later than #9411, but near enough.

Reply:
Should this note be in a comment on this line, or in a `Changes in SCHEMA_COMPAT_VERSION = 60:` section as above?

Reply:
> Changes in SCHEMA_COMPAT_VERSION = 60

well, that would be a bad name for the section, since it's not really a change in the "schema compat version" - I'm not even sure that makes any sense. But it's a fair question to ask if it should be in a bit of a longer comment rather than a one-liner here.

What I'm trying to avoid is the situation where somebody bumps the number and forgets to update the comment. I figure if they are on the same line, you have to try extra hard to mess it up. But it does only leave me with about 50 characters, which makes for a rather terse comment. 

We could go with a separate comment and *also* a "# remember to update the comment above" comment, but that feels kinda clunky too.

So, open to suggestions/discussion.

Reply:
I was mostly just worried about there only being room for a single note here, which would get overwritten in the future when other changes are made.

But are historical changes not a concern here?

Reply:
Maybe we could match the style of the other comment, but come up with a more accurate heading?

```
Changes leading to SCHEMA_COMPAT_VERSION = 60:
    - "outlier" is no longer stored in event_json.internal_metadata for backwards compatibility
```

I do worry that putting a comment on the line has led you to making it a bit terse that I didn't know what you meant until reading the code.

However, wouldn't mind if you thought that we may well not care about the history and the presence of a comment may not particularly matter — git blame can point you to where things are changed.

Reply:
> But are historical changes not a concern here?

well no, not really. Once we dropped support for synapses with `SCHEMA_VERSION == 60`, we'll no longer really care about the reasons we dropped support for synapses with `SCHEMA_VERSION == 59`.  It's the same reason we don't keep a complete history in https://github.com/matrix-org/synapse/blob/develop/synapse/python_dependencies.py: as long as we don't support Twisted 16, who the hell cares why we don't support Twisted 14?

Reply:
maybe we don't need a comment here at all. After all, there's always git. I just thought it might be helpful for the curious.